http-client/src/test/java/io/airlift/http/client/AbstractHttpClientTest.java

```diff
@@ -877,104 +906,82 @@ public void testDowngradedHttpProtocolUsed()
-                        {
-                            @Override
-                            public Integer handleException(Request request, Exception exception)
-                                    throws RuntimeException
-                            {
-                                throw propagate(request, exception);
-                            }
+                        case "/pipe" -> {
+                            // pipe test is here:
+                            // execute a request to get content and then use the input stream as an argument
+                            // to stream another put request
 
-                            @SuppressWarnings("StatementWithEmptyBody")
-                            @Override
-                            public Integer handle(Request request, Response response)
-                                    throws RuntimeException
-                            {
-                                try {
-                                    InputStream in = response.getInputStream();
-                                    while (in.read() >= 0) {
-                                        // do nothing
-                                    }
-                                }
-                                catch (IOException e) {
-                                    throw new UncheckedIOException(e);
-                                }
-                                return response.getStatusCode();
+                            Request pipeRequest = prepareGet().setUri(locals.server.baseURI().resolve("get")).build();
+                            StreamingResponse streamingResponse = httpClient.executeStreaming(pipeRequest);
+
+                            Request putRequest = preparePut().setUri(locals.server.baseURI().resolve("put")).setBodyGenerator(streamingBodyGenerator(streamingResponse.getInputStream())).build();
+                            try {
+                                executeRequest(locals.server, putRequest, createStatusResponseHandler());
                             }
-                        };
-
-                        int statusCode = executeRequest(server, streamingRequest, testResponseHandler);
-                        assertThat(statusCode).isEqualTo(200);
-                        assertThat(server.servlet().getRequestMethod()).isEqualTo("PUT");
-                        assertThat(server.servlet().getRequestUri()).isEqualTo(uri);
-                        assertThat(server.servlet().getRequestHeaders(CONTENT_TYPE)).isEqualTo(ImmutableList.of("x-test"));
-                        assertThat(server.servlet().getRequestBytes()).isEqualTo(content.getBytes());
-                    }
-                    catch (Exception e) {
-                        throw new RuntimeException(e);
-                    }
+                            catch (Exception e) {
+                                throw new RuntimeException(e);
```

Comment:
```suggestion
                            catch (IOException e) {
                                throw new UncheckedIOException(e);
```

Reply:
The test's abstract `executeRequest()` throws Exception.